website/addons/googledrive/model.py

```diff
@@ -111,218 +101,183 @@ def get_or_create(cls, node, path):
-    username = fields.StringField()
-    access_token = fields.StringField()
-    refresh_token = fields.StringField()
-    expires_at = fields.DateTimeField()
+    # TODO: Remove, if not used
+    def _get_folders(self):
+        """ Get a list of a user's folders"""
+        client = self._drive_client
+        return client.folders()
+
+    def _refresh_token(self, access_token, refresh_token):
+        client = self._auth_client
+        if refresh_token:
+            token = client.refresh(access_token, refresh_token)
+            return token
+        else:
+            exceptions.AddonError("Refresh Token is not Obtained")
 
     def fetch_access_token(self, force_refresh=False):
         self.refresh_access_token(force=force_refresh)
-        return self.access_token
+        return self.account.oauth_key
 
     def refresh_access_token(self, force=False):
         if self._needs_refresh() or force:
-            client = GoogleAuthClient()
-            token = client.refresh(self.access_token, self.refresh_token)
-
-            self.access_token = token['access_token']
-            self.refresh_token = token['refresh_token']
-            self.expires_at = datetime.utcfromtimestamp(token['expires_at'])
-            self.save()
-
-    def revoke_access_token(self):
-        # if there is only one osf user linked to this google drive user oauth, revoke the token,
-        # otherwise, disconnect the osf user from the googledriveoauthsettings
-        if len(self.googledriveusersettings__accessed) <= 1:
-            client = GoogleAuthClient()
-            try:
-                client.revoke(self.access_token)
-            except:
-                # no need to fail, revoke is opportunistic
-                pass
-
-            # remove the object as its the last instance.
-            GoogleDriveOAuthSettings.remove_one(self)
+            token = self._refresh_token(self.account.oauth_key, self.account.refresh_token)
+            self.account.oauth_key = token['access_token']
+            self.account.refresh_token = token['refresh_token']
+            self.account.expires_at = datetime.utcfromtimestamp(token['expires_at'])
+            self.account.save()
 
     def _needs_refresh(self):
-        if self.expires_at is None:
+        if self.account.expires_at is None:
             return False
-        return (self.expires_at - datetime.utcnow()).total_seconds() < drive_settings.REFRESH_TIME
+        return (self.account.expires_at - datetime.utcnow()).total_seconds() < drive_settings.REFRESH_TIME
 
 
-class GoogleDriveUserSettings(AddonUserSettingsBase):
-    """Stores user-specific information, including the Oauth access
-    token.
-    """
-    oauth_settings = fields.ForeignField(
-        'googledriveoauthsettings', backref='accessed'
-    )
+class GoogleDriveUserSettings(StorageAddonBase, AddonOAuthUserSettingsBase):
+    oauth_provider = GoogleDriveProvider
+    oauth_grants = fields.DictionaryField()
```

Comment:
This field overrides an identically named field on the parent class, which seems unnecessary.